docs/index.rst

```diff
@@ -0,0 +1,1032 @@
+    </dependency>
+
+You can also download the development version of the bindings directly
+from the Git repository
+
+.. code-block:: bash
+
+    $ git clone git://github.com/bigmlcom/bigml-java.git
+
+Authentication
+--------------
+
+All the requests to BigML.io must be authenticated using your username
+and `API key <https://bigml.com/account/apikey>`_ and are always
+transmitted over HTTPS.
+
+This module will look for your username and API key in the
+``binding.properties`` file. Alternatively, you can respectively set
+the JVM parameters ``BIGML_USERNAME`` and ``BIGML_API_KEY``  with ``-D``.
+
+With that set up, connecting to BigML is a breeze.
+First, import ``BigMLClient``:
+
+.. code-block:: java
+
+    import org.bigml.binding.BigMLClient;
+
+then:
+
+.. code-block:: java
+
+    BigMLClient api = BigMLClient.getInstance();
+
+Otherwise, you can initialize directly when instantiating
+the BigMLClient class as follows:
+
+.. code-block:: java
+
+    BigMLClient api = BigMLClient.getInstance("myusername",
+        "ae579e7e53fb9abd646a6ff8aa99d4afe83ac291");
+
+Also, you can initialize the library to work in the Sandbox
+environment by passing the parameter ``dev_mode``:
+
+.. code-block:: java
+
+    BigMLClient api = BigMLClient.getInstance(true);
+
+or:
+
+.. code-block:: java
+
+    BigMLClient api = BigMLClient.getInstance("myusername",
+        "ae579e7e53fb9abd646a6ff8aa99d4afe83ac291", true);
+
+For Virtual Private Cloud setups, you can change the remote server URL
+to the VPC particular one by either setting the ``BIGML_URL`` or
+``BIGML_DEV_URL`` in ``binding.properties`` or in the JVM environment.
+For example:
+
+.. code-block:: java
+
+    BIGML_URL=https://bigml.io/andromeda/
+    BIGML_DEV_URL=https://bigml.io/dev/andromeda/
+
+The corresponding SSL REST calls will be directed to your private domain
+henceforth.
+
+Quick Start
+-----------
+
+Imagine that you want to use `this csv
+file <https://static.bigml.com/csv/iris.csv>`_ containing the `Iris
+flower dataset <http://en.wikipedia.org/wiki/Iris_flower_data_set>`_ to
+predict the species of a flower whose ``sepal length`` is ``5`` and
+whose ``sepal width`` is ``2.5``. A preview of the dataset is shown
+below. It has 4 numeric fields: ``sepal length``, ``sepal width``,
+``petal length``, ``petal width`` and a categorical field: ``species``.
+By default, BigML considers the last field in the dataset as the
+objective field (i.e., the field that you want to generate predictions
+for).
+
+::
+
+    sepal length,sepal width,petal length,petal width,species
+    5.1,3.5,1.4,0.2,Iris-setosa
+    4.9,3.0,1.4,0.2,Iris-setosa
+    4.7,3.2,1.3,0.2,Iris-setosa
+    ...
+    5.8,2.7,3.9,1.2,Iris-versicolor
+    6.0,2.7,5.1,1.6,Iris-versicolor
+    5.4,3.0,4.5,1.5,Iris-versicolor
+    ...
+    6.8,3.0,5.5,2.1,Iris-virginica
+    5.7,2.5,5.0,2.0,Iris-virginica
+    5.8,2.8,5.1,2.4,Iris-virginica
+
+You can easily generate a prediction following these steps:
+
+.. code-block:: java
```

Comment:
this one looks the only complete code example and this is because I think that showing how to implement a simple main would be useful, we had users in the past asking for that simple example. It's fine how the other snippets are listed, but this one is worth to have it complete. At least a link to the BigMLSmapleClient.java ...